src/avram/credentials.cr

```diff
@@ -0,0 +1,125 @@
+class Avram::Credentials
+  getter url : String = ""
+
+  def initialize(
+    @database : String,
+    @hostname : String? = nil,
+    @username : String? = nil,
+    @password : String? = nil,
+    @port : Int32? = nil,
+    @query : String? = nil
+  )
+  end
+
+  def self.void : Credentials
+    build(database: "unused")
+  end
+
+  def self.build(**args) : Credentials
+    new(**args).build
+  end
+
+  def self.parse(nothing : Nil)
+    nil
+  end
```

Comment:
Could this be removed? I'm worried this will lead to accidental errors or confusion where people try to parse `Nil`

Reply:
The issue there is if you do `Credentials.parse(ENV["NOTHING"]?)`, but that method only takes a string, then it would always fail and not allow you to fallback to building the URL the other way. I guess we could just generate apps only using the manual builder, and add a comment that says something like "If you want to use an ENV var, use .parse" 🤔 

Let me know if that makes sense.